pkg/ca/googleca/v1/googleca.go

```diff
@@ -136,32 +136,40 @@ func Req(parent string, pemBytes []byte, cert *x509.Certificate) (*privatecapb.C
 	}, nil
 }
 
-func (c *CertAuthorityService) Root(ctx context.Context) ([]*x509.Certificate, error) {
-	c.cachedRootsOnce.Do(func() {
-		var pems string
-		var err error
-		cas := c.client.ListCertificateAuthorities(ctx, &privatecapb.ListCertificateAuthoritiesRequest{
-			Parent: c.parent,
-		})
-		for {
-			c, done := cas.Next()
-			if done == iterator.Done {
-				break
-			}
-			if done != nil {
-				break
-			}
-			pems += strings.Join(c.PemCaCertificates, "")
-		}
-		c.cachedRoots, err = cryptoutils.LoadCertificatesFromPEM(strings.NewReader(pems))
-		if err != nil {
-			panic(fmt.Errorf("failed parsing PemCACertificates: %w", err))
-		}
+func (c *CertAuthorityService) TrustBundle(ctx context.Context) ([][]*x509.Certificate, error) {
+	// if we've already successfully fetched the CA info, just use the cached value
+	if c.cachedRoots != nil {
+		return [][]*x509.Certificate{c.cachedRoots}, nil
+	}
+
+	// fetch the latest values for the specified CA
+	var pems string
+	cas := c.client.ListCertificateAuthorities(ctx, &privatecapb.ListCertificateAuthoritiesRequest{
+		Parent: c.parent,
 	})
-	if len(c.cachedRoots) == 0 {
-		return c.cachedRoots, fmt.Errorf("error fetching root certificates")
+	for {
+		ca, done := cas.Next()
+		if done == iterator.Done {
+			break
+		} else if done != nil {
+			// if the iterator returns an issue for some reason, exit
+			return [][]*x509.Certificate{}, done
+		}
+		pems += strings.Join(ca.PemCaCertificates, "")
 	}
-	return c.cachedRoots, nil
+	// if we fail to parse the PEM content, return an error
+	roots, err := cryptoutils.LoadCertificatesFromPEM(strings.NewReader(pems))
+	if err != nil {
+		return [][]*x509.Certificate{}, fmt.Errorf("failed parsing PemCACertificates response: %w", err)
+	}
+	if len(roots) == 0 {
+		return [][]*x509.Certificate{}, fmt.Errorf("error fetching root certificates")
+	}
+	c.cachedRootsOnce.Do(func() {
```

Comment:
Does this lock execution for other threads? As in, will other threads block until `Once` is done, so `c.cachedRoots` will always be populated by the end of this? I assume yes, but just checking.

Reply:
Looks like yep, there's a comment with this case - https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/sync/once.go;l=49-55